src/pages/app-bar/contextual-action/index.tsx

```diff
@@ -0,0 +1,314 @@
+    },
+    deleteRow: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        marginBottom: `${theme.spacing(2)}px`,
+    },
+    noResult: {
+        padding: `${theme.spacing(2)}px`,
+        textAlign: 'center',
+        backgroundColor: theme.palette.background.paper,
+    },
+    noteText: {
+        color: colors.gray[500],
+        marginTop: `${theme.spacing(3)}px`,
+    },
+    resetTableLink: {
+        textDecoration: 'underline',
+        color: colors.blue[500],
+        cursor: 'pointer',
+    },
+    rowSelected: {
+        backgroundColor: `rgba(${colors.blue[500]}, 0.05)`,
+    },
+    sticky: {
+        position: 'sticky',
+        left: 0,
+    },
+    secondaryToolbar: {
+        backgroundColor: colors.black[500],
+        position: 'absolute',
+        width: '100%',
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    tableBody: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    tableContainer: {
+        overflow: 'auto',
+        maxWidth: '800px',
+        width: '100%',
+        padding: `${theme.spacing(2)}px`,
+        [theme.breakpoints.down('sm')]: {
+            maxWidth: 'unset',
+            padding: 0,
+        },
+    },
+    toolbarGutters: {
+        paddingLeft: 16,
+        paddingRight: 4,
+    },
+}));
+
+const createItem = (index: number, ip: string): ListItemType => ({
+    id: index,
+    name: `Device 0${index}`,
+    ip: ip,
+    checked: false,
+});
+
+const generatedList: ListItemType[] = [];
+
+for (let i = 1; i < 5; i++) {
+    generatedList.push(createItem(i, '192.168.0.1'));
+}
+
+export const ContextualAction = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const theme = useTheme();
+    const classes = useStyles();
+    const [list, setList] = useState<ListItemType[]>(generatedList);
+    const [selectedItems, setSelectedItems] = useState<ListItemType[]>([]);
+    const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+    const onSelect = useCallback(
+        (item: ListItemType): void => {
+            if (!selectedItems.includes(item)) {
+                setSelectedItems([...selectedItems, item]);
+            } else {
+                const index = selectedItems.indexOf(item);
+                setSelectedItems(selectedItems.filter((_: ListItemType, i: number) => i !== index));
+            }
+        },
+        [selectedItems, setSelectedItems]
+    );
+
+    const isSelected = useCallback((item: ListItemType): boolean => selectedItems.includes(item), [selectedItems]);
+
+    const onDelete = useCallback((): void => {
+        const updatedList = [...list];
+
+        selectedItems.forEach((item: ListItemType) => {
+            const index = updatedList.indexOf(item);
+            updatedList.splice(index, 1);
+        });
+
+        setList(updatedList);
+        setSelectedItems([]);
+    }, [list, selectedItems, setList, setSelectedItems]);
```

Comment:
`setList` and `setSelectedItems` will never change so no need to have this in the dependencies.

Reply:
Done please review